client/components/banner/test/test.jsx

```diff
@@ -0,0 +1,201 @@
+/** @format */
+
+jest.mock( 'lib/abtest', () => ( {
+	abtest: () => '',
+} ) );
+
+jest.mock( 'blocks/dismissible-card', () => {
+	const React = require( 'react' );
+	return class DismissibleCard extends React.Component {};
+} );
+
+jest.mock( 'components/card', () => {
+	const React = require( 'react' );
+	return class Card extends React.Component {};
+} );
+
+jest.mock( 'lib/analytics/track-component-view', () => {
+	const React = require( 'react' );
+	return class TrackComponentView extends React.Component {};
+} );
```

Comment:
Unfortunate to need to mock these components even in a `shallow` render 😞 

These components are "nice" for readability in the tests (`find( 'Card' )`), as an alternative you can use something like this:

https://github.com/Automattic/wp-calypso/blob/6adb9972a617f84ee91f11ba174e0dc70173c8f9/client/components/theme/test/index.jsx#L23-L24

It's a bit less ceremony/overhead, but names can't correspond exactly to these camel-cased names.